### DISCUSSION TITLE
New release slaps!

### DISCUSSION THREAD

**Discussion by CreaturesFarley** [shells-dw/streamdeck-totalmix#18] [General]
Just noticed my plugin has updated, with a slight graphical update. The new version works like a charm. No latency. Thanks for all your work on this! This is my most used StreamDeck plugin by far!

**Comment by shells-dw** [comment]
Thanks for your feedback buddy! Much appreciated ♥️
I'm glad it helps you  😃 
Be sure to let me know in case you encounter any issues.

I've been continuing developing it since a couple of months again and reworked the whole OSC communications in the background - which is something I wanted to do since quite a while, the old implementation turned out to be slow and buggy and for some reason Windows 11 didn't quite like it... or vice versa. I never tried to figure out why, but once I finally upgraded to Win 11 myself I could at least verify it's actually an issue. While at it, I improved a bunch of other things with it, (including some states where the plugin could eat up a whole CPU core while not really doing anything useful...) be sure to check out the [changelog in the StreamDeck Store](https://apps.elgato.com/plugins/de.shells.totalmix) if you haven't already, maybe there are other things you find useful as well.

Anyway, with the basically complete rewrite of the device communication it took me a while to get something that reliably works. Publishing a plugin with Elgato can take a while and given quite a few people downloaded and probably use the plugin for work every day, I was rather hesitant to publish it until I was as sure as I could be that it works reliably now.

Which is just a long way of saying... sub this repo here if you want the newest releases (in case there is something changing again) and maybe participate in some beta testing.
Right now I'm kind of happy how it is, but if there is something changed in the future, it'll be here first.

[edit] aaaaaaaand just noticed you were already aware of this repo, we had a conversation a couple of weeks ago about an issue - woops, sorry 😅 [/edit]

Thanks again.
Cheers!

**Reply by Misoh126** [reply]
hey
I use the plugin on my AIO Pro and found that with the new update, the channels are no longer recognized.
The field for mirror also no longer appears.
Everything worked with unchanged settings, but new buttons cannot be set up.
It would be nice if you could check it out again.

Many Thanks :)

**Reply by shells-dw** [reply]
Hi Misoh,

did you refer to the readme here on GitHub? Starting [here](https://github.com/shells-dw/streamdeck-totalmix#updating-from-older-versions) until the MIDI part starts should provide you with all information you need to update.

I'm sorry for the inconvenience caused by the update. I've basically rewritten most of the plugin to make it more robust and eliminate issues it undoubtably had (especially basically not working with Windows 11 was kind of a showstopper) - which might sadly break existing installations, but due to how StreamDeck and the update process works, there is not much that can be done about that. EIther I push a new version or I don't and that's that. There is not a nice way to show users a prompt in the lines of "hey, there was an update, look what changed and look what you have to do to make it work". Additionally with the requirement to enable the second OSC listener in TotalMix to improve the mirroring, that's nothing I want do programmatically - so I approved potential issues like these for the greater good 😉  

Long story short: A friend of mine uses an HDSPe RayDAT and that works fine with the new version. I suspect something in the setup is still missing or gone wrong.
Check the readme.
If any button acts up, remove and replace it. Button settings and part of the logic are stored with a button at the time it is put on the StreamDeck and never updated and when the underlaying plugin code changes, it can break the button. That's how it is in the StreamDeck 🤷 .
If you're confident you've set everything up correctly according to the [readme](https://github.com/shells-dw/streamdeck-totalmix#updating-from-older-versions) and it still doesn't work, please elaborate on the actual issues you're facing - "new buttons cannot be set up", what does that mean?

Good luck 😃

**Reply by Misoh126** [reply]
Many thanks for your response!
I followed you and followed the instructions.
I also found the error!
The error was that from version 3.0.0 2 ports are used and only one was active for me.

I will always read the readme first now :)

I just think it's a pity that the remote network function has been removed.
But that's complaining on a high level.

I say thank you for taking the time for the plugin!
And sorry for the wasted time on me :)

**Reply by shells-dw** [reply]
No worries buddy 😃 

Glad it's working again for you!

> I just think it's a pity that the remote network function has been removed. But that's complaining on a high level.
> 
well, it has not been. You can set the target IP in the config file - as described in the [readme](https://github.com/shells-dw/streamdeck-totalmix#deshellstotalmixexeconfig). The value assigned to  ` <add key="interfaceIp" value="127.0.0.1" />` is the IP of the PC in your network that runs TotalMix.

What isn't possible anymore is having different buttons sending commands to different TotalMix instances in the network (or say different PCs with connected interfaces). I never came across such a setup (and mirroring multiple targets wouldn't be really feasible, but that's a different story).
Is that actually something you need? Do you run multiple PCs with TotalMix and connected interfaces?
